src/regularized/abstract_regularized.jl

```diff
@@ -0,0 +1,27 @@
+"""
+    AbstractRegularized{parallel} <: AbstractOptimizationLayer
+
+Convex regularization perturbation of a black box optimizer
+```
+ŷ(θ) = argmax_{y ∈ C} {θᵀy - Ω(y)}
+```
+
+# Interface
+
+- `(regularized::AbstractRegularized)(θ; kwargs...)`: return `ŷ(θ)`
+- `compute_regularization(regularized, y)`: return `Ω(y)`
+
+# Available implementations
+
+- [`SoftArgmax`](@ref)
+- [`SparseArgmax`](@ref)
+- [`RegularizedFrankWolfe`](@ref)
```

Comment:
Should we also provide another implementation allowing the user to provide its own optimizer, or is subtyping from `AbstractRegularized` enough ?

Reply:
I think subtyping is actually easier, cause they're gonna need two methods anyway (callable and get regularization)

Reply:
Then we should explain somehere (in the doc ?) how to create a custom regularized

Reply:
Yep, and test it too. But this can come after the release in a patch